src/Yaapii.Atoms/IO/Zip.cs

```diff
@@ -49,15 +50,24 @@ public Zip(string path)
     public Stream Stream()
     {
         AssumeIsDirectory(this.path);
+
+        var directories = Directory.GetFiles(this.path, "*", SearchOption.AllDirectories);
+        var array = new string[directories.Length];
+        for (int i = 0; i < directories.Length; i++)
+        {
+            array[i] = Path.GetFileName(directories[i]);
```

Comment:
This will lead to problems. All files, beginning from the given root are zipped in this class. Now, only the filenames are zipped without directory structure? What if the files under ```path``` are the following:
- ./A.txt
- ./dir1/A.txt
- ./dir2/A.txt

a zip entry with 3 entries "A.txt" is not valid.